sdk/ai/Azure.AI.Projects/src/Custom/AIProjectClient.cs

```diff
@@ -2,23 +2,61 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Linq;
-using Azure.AI.Inference;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
 using Azure.Core;
+using Azure.Identity;
 
 namespace Azure.AI.Projects
 {
     // Data plane generated client.
     /// <summary> The AzureAI service client. </summary>
-    public partial class AIProjectClient
+    public partial class AIProjectClient : ConnectionProvider
     {
+        /// <summary>
+        /// Protects the <see cref="Connections"/> collection from concurrent access. Separated from <see cref="_connectionCacheLock"/> to reduce contention.
+        /// </summary>
+        private readonly ReaderWriterLockSlim _connectionsLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+
+        /// <summary>
+        /// Protects the <see cref="_connectionCache"/> dictionary from concurrent access. Separated from <see cref="_connectionsLock"/> to improve concurrency in read-heavy scenarios.
+        /// </summary>
+        private readonly ReaderWriterLockSlim _connectionCacheLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+
+        /// <summary>
+        /// Subclient connections.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ConnectionCollection Connections { get; } = [];
```

Comment:
why does it have to be public? Should all public code call GetConnection on the provider?

Reply:
Removed